src/background/telemetry/app-insights-telemetry-client.ts

```diff
@@ -36,7 +36,7 @@ export class AppInsightsTelemetryClient implements TelemetryClient {
         this.updateTelemetryState();
     }
 
-    public trackEvent(name: string, properties?: { [name: string]: string }): void {
+    public async trackEvent(name: string, properties?: { [name: string]: string }): Promise<void> {
         if (this.enabled) {
             this.applicationInsights.trackEvent({ name }, properties);
```

Comment:
My understanding is that this is synchronous because it adds the telemetry to a background batching/queuing system. I'm not sure that's a great idea for us, since we might lose telemetry that way

Reply:
(but I'd also be very hesitant to make it flush every time and await `trackEvent` every time without being very careful about the order in which we're invoking `trackEvent` vs other business logic - making sure the service worker lifetime waits for the network event would be good, but network calls can be arbitrarily slow, I wouldn't want the app responsiveness to ever be awaiting for telemetry to flush)

Reply:
The original issue was that trackEvent in DebugToolsTelemetryClient sends a runtime message, and making trackEvent async there required a change to the TelemetryClient interface that affected everything that implements it. It sounds like changing the other telemetry clients could create more problems than keeping DebugToolsTelemetryClient fire-and-forget, and I'd rather we lose messages in the debug tools than in app insights, so I'll revert this change.

Reply:
Per offline discussion, this is an existing issue in the extension which is not affected by whether or not this function is async. A potential solution is to flush app insights telemetry regularly in the background, for instance each time we receive a message. This will be addressed in a later PR.